components/collator/src/comparison.rs

```diff
@@ -100,30 +98,24 @@ impl Collator {
             // the default, the code here that omits the explicit
             // variant if it matches the default should become
             // unnecessary.
-            let original_locale: Locale = locale.into();
-            let mut filtered_locale: Locale = original_locale.id.into();
+            let original_locale = locale;
```

Comment:
When I made the change to adopt Cow, I figured there was about an 80% chance that you would reply with a comment saying that I should improve the block expression. I didn't do it because this is internal implementation code and it doesn't really matter, and I have 4 other PRs in flight and not enough time to focus on making each one perfect today.

Reply:
> I didn't do it because this is internal implementation code and it doesn't really matter

We still want to keep our internal implementation code readable. The next person will have to go figure out why the locale is renamed two times.

Reply:
I agree; #2314